pkg/kv/kvserver/client_raft_test.go

```diff
@@ -6206,13 +6206,49 @@ func TestRaftPreVote(t *testing.T) {
 				tc.WaitForValues(t, key, []int64{2, 2, 2})
 				t.Logf("n1 has lease")
 
-				// Wait for the range to quiesce, if enabled.
+				// Block new proposals to the range.
+				blocked.Store(true)
+				t.Logf("n1 proposals blocked")
+
+				// Wait for the range to quiesce, if enabled. Otherwise, wait for the
+				// range to stabilize such that all replicas have the same last and
+				// commit index for a second -- similarly to the quiesce checks.
 				if quiesce {
 					require.Eventually(t, repl3.IsQuiescent, 10*time.Second, 100*time.Millisecond)
 					t.Logf("n3 quiesced")
 				} else {
 					require.False(t, repl3.IsQuiescent())
 					t.Logf("n3 not quiesced")
+
+					var lastIndex uint64
+					var lastChanged time.Time
+					require.Eventually(t, func() bool {
```

Comment:
Is there a simpler condition to wait for? For example, all commit indices match the leader's last index. This will imply all the rest, e.g. that `Match` matches on all the nodes.

Or this would be racy? I suspect that things can be flushed to raft in the meantime, so the last index can grow for a bit.

Reply:
We can drop the commit index check, it's sort of implied by the match checks. Looking at the commit index requires looking at the local replica state across all replicas, seems easier to just look at the leader's state.